### DISCUSSION TITLE
Progressive Registration (or User Journeys)

### DISCUSSION THREAD

**Discussion by da2ce7** [torrust/torrust-index-backend#306] [Feature requests]
Let the user registration and privilege granting process be progressive.

Such stages might include:

- Public (no account).
- Anon (no username)
- Name (username, no password)
- User (username, password)
- Mail (with email)
- Photo
- Bio
- Full Account
- Moderator
- Administrator

Our system should be able to keep track of the progressive lifetime of the user.

**Comment by josecelano** [comment]
Relates to:

- https://github.com/torrust/torrust-index/issues/507
- https://github.com/torrust/torrust-index-gui/issues/287

**Comment by josecelano** [comment]
Hi @da2ce7, I'm working on the roadmap. I will add some feature issues for this EPIC and link them here.
